tests/identities/dual_identity_d1.prot

```diff
@@ -0,0 +1,42 @@
+struct DualIdentity {
+  in a: u32,
+  in b: u32,
+  out s1: u32,
+  out s2: u32,
+}
+
+fn one<dut: DualIdentity>(in a: u32, out s1: u32, out s2 : u32) {
+    dut.a := 1;
+    dut.b := X;
+    fork();
+
+    step();
+
+    if (dut.a == a) {
```

Comment:
can you add a comment here to explain why this `if` construct here is relevant for your test?
on its own, this protocol is not self-explanatory.